controller/src/beerocks/master/wfa_ca.cpp

```diff
@@ -11,8 +11,32 @@
 
 #include <algorithm>
 
+#define FORBBIDEN_CHARS "$#&*()[]{};:,/\\<>?|`~=+"
+
 using namespace beerocks;
 
+static std::size_t g_token_idx = 0;
+static std::vector<std::string> g_cmd_tokens_vec;
+
+inline static const std::string &get_next_token() { return g_cmd_tokens_vec[g_token_idx++]; }
+inline static bool finished_tokens()
+{
+    return g_token_idx == g_cmd_tokens_vec.size();
+    LOG(DEBUG) << "Tokens_num=" << int(g_cmd_tokens_vec.size())
+               << ", g_token_idx=" << int(g_token_idx);
+}
```

Comment:
It is global in a CPP file that runs on a single thread.
I think it is cleaner how I did it.

Reply:
I strongly disagree that it is cleaner. There is simply no reason to use globals in this case, except to avoid having to pass an argument to a function.

Reply:
fine I changed it. But I really believe that what I did looks better :\

Reply:
I strongly agree with Arnout on this one. Globals are evil and should be avoided if not absolutely necessary 😉